conan/cli/commands/remote.py

```diff
@@ -201,10 +201,13 @@ def remote_login(conan_api, parser, subparser, *args):
     for r in remotes:
         previous_info = conan_api.remotes.user_info(r)
 
-        user, password = creds.auth(r, args.username, args.password)
-        if args.username is not None and args.username != user:
-            raise ConanException(f"User '{args.username}' doesn't match user '{user}' in "
-                                 f"credentials.json or environment variables")
+        if args.username is not None and args.password is not None:
```

Comment:
This is the main change. I removed the ``password`` argument in the ``creds.auth()``, it was confusing and it made you add it in your PR, because I think the argument was not right in the first place